compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -2149,11 +2148,13 @@ object desugar {
       enums match {
         case Nil if betterForsEnabled => body
         case (gen: GenFrom) :: Nil =>
+          val aply = Apply(rhsSelect(gen, mapName), makeLambda(gen, body))
           if betterForsEnabled
-            && gen.checkMode != GenCheckMode.Filtered // results of withFilter have the wrong type
-            && deepEquals(gen.pat, body)
-          then gen.expr  // avoid a redundant map with identity
-          else Apply(rhsSelect(gen, mapName), makeLambda(gen, body))
+          && gen.checkMode != GenCheckMode.Filtered // results of withFilter have the wrong type
```

Comment:
This is non-standard indentation, but I'd tend t leave it in to be able to judge better how it looks.